### DISCUSSION TITLE
Add flag to disable chat sharing

### DISCUSSION THREAD

**Discussion by tkg61** [open-webui/open-webui#6377] [Ideas]
# Feature Request

Could we add a environment flag or UI flag to disable the share button on chats? We would like to limit users sharing chats which might contain sensitive information. I know tickets #2890 and also #2904 are around sharing and managing it but they don't mention turning off sharing altogether from an admin perspective. We jus would like to remove the button altogether and the "disable community sharing" does not remove this either.

**Describe the solution you'd like**
Add a env flag or UI flag to disable the sharing chat feature

**Describe alternatives you've considered**
N/A

**Comment by silentoplayz** [comment]
`Enable Community Sharing` is likely the option you're looking for to be toggled off in the UI. This option can be found within the Admin Panel settings within the `General` tab.
![image](https://github.com/user-attachments/assets/6494c95a-eb61-46bb-8607-d31129f7fd57)

If you would like to set an environment variable to toggle this functionality off instead, refer to the documentation for [ENABLE_COMMUNITY_SHARING](https://docs.openwebui.com/getting-started/env-configuration#enable_community_sharing).

While this doesn't prevent the `Share` option from being displayed for chats, it does prevent chats from being shared on the community platform (see https://openwebui.com/prompts) from that share button, limiting the share option to a chat link that can be shared. Only users with an account and logged into the Open WebUI instance are able to view such shared chats.

Here's what you'll see when attempting to share a chat with `Enable Community Sharing` toggled off:
![image](https://github.com/user-attachments/assets/4158f5c8-9ea4-4a4d-92f8-5bd3e1e63fda)

My two cents for a QOL suggestion:
Perhaps `Allow Chat Sharing` would fit well here in the `Admin Panel` settings to truly toggle on/off this chat sharing feature for User accounts:
![image](https://github.com/user-attachments/assets/edbd20a3-c16e-4cfd-b824-2f1f0449832e)
